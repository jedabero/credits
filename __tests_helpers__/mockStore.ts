// eslint-disable-next-line import/no-extraneous-dependencies
import configureMockStore from 'redux-mock-store';
import { RootState } from '../src/store';

export default configureMockStore<RootState>();
