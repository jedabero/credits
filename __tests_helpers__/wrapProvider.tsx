import React, { ReactElement } from 'react';
import { Provider } from 'react-redux';
import { Store } from '../src/store';

const wrapProvider = (component: ReactElement, store: Store) => (
  <Provider store={store}>{component}</Provider>
);

export default wrapProvider;
