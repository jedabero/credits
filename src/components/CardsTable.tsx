import { EntityId } from '@reduxjs/toolkit';
import { selectCreditCardsIds } from '../reducers/creditCards';
import { useAppSelector } from '../hooks';
import CardRow from './CardRow';

type CardsTableProps = {
  onEdit: React.Dispatch<React.SetStateAction<EntityId | undefined>>,
  onView: React.Dispatch<React.SetStateAction<EntityId | undefined>>,
};

function CardsTable({ onEdit, onView }: CardsTableProps) {
  const cardsIds = useAppSelector(selectCreditCardsIds);

  return (
    <div className="cardstable">
      <div className="cardrow">
        <div>ID</div>
        <div>Name</div>
        <div>Number</div>
        <div>Quota</div>
        <div>Balance</div>
        <div>Cut off day</div>
        <div>Actions</div>
      </div>
      {cardsIds.map((id) => (
        <CardRow key={id} cardId={id} onEdit={onEdit} onView={onView} />
      ))}
    </div>
  );
}

export default CardsTable;
