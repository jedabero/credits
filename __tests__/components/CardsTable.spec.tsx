/* eslint-disable import/no-extraneous-dependencies */
import React from 'react';
import { create } from 'react-test-renderer';
import CardsTable from '../../src/components/CardsTable';
import wrapProvider from '../../__tests_helpers__/wrapProvider';
import mockStore from '../../__tests_helpers__/mockStore';

describe('CardsTable test', () => {
  test('should render', () => {
    const store = mockStore({
      _persist: { version: 0, rehydrated: true },
      creditCards: {
        entities: {},
        ids: [],
      },
      movements: {
        entities: {},
        ids: [],
      },
    });
    const tree = create(
      wrapProvider(
        <CardsTable onEdit={() => {}} onView={() => {}} />,
        store,
      ),
    );
    expect(tree.toJSON()).toMatchSnapshot();
  });
});
