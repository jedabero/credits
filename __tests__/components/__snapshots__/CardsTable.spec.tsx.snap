// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CardsTable test should render 1`] = `
<div
  className="cardstable"
>
  <div
    className="cardrow"
  >
    <div>
      ID
    </div>
    <div>
      Name
    </div>
    <div>
      Number
    </div>
    <div>
      Quota
    </div>
    <div>
      Balance
    </div>
    <div>
      Cut off day
    </div>
    <div>
      Actions
    </div>
  </div>
</div>
`;
